samples/shopping-cart-addview/src/main/java/shoppingcart/application/ShoppingCartView.java

```diff
@@ -0,0 +1,82 @@
+package shoppingcart.application;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import akka.javasdk.annotations.ComponentId;
+import akka.javasdk.annotations.Consume;
+import akka.javasdk.annotations.Query;
+import akka.javasdk.view.TableUpdater;
+import akka.javasdk.view.View;
+import shoppingcart.domain.ShoppingCartEvent;
+
+@ComponentId("shopping_cart_view")
+public class ShoppingCartView extends View {
+  @Query("SELECT * FROM shopping_cart_view WHERE cartId = :cartId")
+  public QueryEffect<Cart> getCart(String cartId) {
+    return queryResult();
+  }
+
+  public record Carts(List<Cart> carts) {
+  }
+
+  public record Cart(String cartId, List<Item> items, boolean checkedout) {
+
+    public Cart addItem(String itemId, String name, int quantity, String description) {
+      var newItems = items;
+      newItems.add(new Item(itemId, name, quantity, description));
+
+      return new Cart(cartId, newItems, false);
+    }
+
+    public Cart withCartId(String nCartId) {
+      return new Cart(nCartId, items, checkedout);
+    }
+
+    public Cart removeItem(String itemId) {
+      var newItems = items;
+      newItems.removeIf(i -> i.itemId() == itemId);
+
+      return new Cart(cartId, newItems, false);
+    }
+
+    public Cart checkout() {
+      return new Cart(cartId, items, true);
+    }
+
+    public record Item(String itemId, String name, int quantity, String description) {
+    }
+
+  }
+
+  @Consume.FromEventSourcedEntity(ShoppingCartEntity.class)
+  public static class CartsTable extends TableUpdater<Cart> {
+
+    @Override
+    public Cart emptyRow() {
+      return new Cart("", new ArrayList<Cart.Item>(), false);
```

Comment:
```suggestion
      return new Cart(updateContext().eventSubject().get(), new ArrayList<Cart.Item>(), false);
```

Since 3.2.2, you can access the updateContext from inside `emptyRow`. 
The event subject is an optional field, but it will always be filled with the entity id when subscribing to a ESE.

Reply:
oh that's excellent. I'll definitely change that.